### PR TITLE
Method calls on P/C

### PR DESCRIPTION
**Pull request by rambleraptor**
We've got a bunch of format + if/else logic to handle this particular case:

What if I have a method call that needs 3 arguments in one case + just 2 in all other cases? What if adding this argument sometimes causes us to go > 100 characters? What if it's 4 arguments and 2 are optional?

Ansible solved this with the `method_decl` and `method_call` functions: Ruby functions that take in a method name and list of arguments and return either a function declaration or a method call. These functions can be given nil arguments that will be ignored and the functions handle all of the formatting issues (what if it's over 100 characters). Formatting function calls is pretty easy for a computer to do.

This is mostly a no-op change. There will be a couple of consistent P/C changes, but they're just formatting changes.

-----------------------------------------------------------------
# [all]

### CODE REVIEW

File: provider/core.rb [hunk 1]
```diff
@@ -461,9 +461,32 @@ def emit_method(name, args, code, file_name, opts = {})
     end
 
     def method_decl(name, args)
```
Comment (ndmckinley) [inline provider/core.rb]: The fact that this happens to work for ruby and python is awesome - go ahead and document that, though?  For instance, if anyone ever needed to add kwargs they'd be annoyed to find out that they broke (puppet & chef) | (ansible).

**Comment by rambleraptor** [comment]
The functions are defined twice.

Once in provider/ansible.rb (Python) and once in provider/core.rb (Ruby).
Depending on which provider you use, it’ll call the proper functions

**Comment by ndmckinley** [comment]
I don't know what to call that, I guess it's polymorphism of the provider?  My question is, is it used for anything?  If not, is it worth a rename to `method_decl_ruby` and `method_decl_python`?

**Comment by rambleraptor** [comment]
I wouldn't call it polymorphism on the provider. I would call it a method override.

The "default" method in `provider/core.rb` is the Ruby one, but Ansible overrides it with a Python version.

We've run into this issue enough that we should discuss a better solution. `provider/core.rb` is filled with Ruby-exclusive functions.

**Comment by ndmckinley** [comment]
Yeah, that's what I think also.  Can you pull this out of `core.rb`, then?

**Pull request merged**